core/templates/dev/head/pages/topic_editor/subtopics_editor/SubtopicsListTabDirective.js

```diff
@@ -27,97 +27,158 @@ oppia.directive('subtopicsListTab', [
-            $scope.subtopicPage = TopicEditorStateService.getSubtopicPage();
-            $scope.subtopicDisplayed = true;
-            $scope.subtopicEditorIsShown = false;
-            $scope.htmlData = $scope.subtopicPage.getHtmlData();
-          });
+                  $scope.closeSubtopicTitleEditor = function() {
+                    $scope.subtopicTitleEditorIsShown = false;
+                  };
+
+                  $scope.openPreviewSubtopicPage = function(htmlData) {
+                    $scope.subtopicEditorIsShown = false;
+                    $scope.htmlData = htmlData;
+                  };
+
+                  $scope.closePreviewSubtopicPage = function(previewHtmlData) {
+                    $scope.subtopicEditorIsShown = true;
+                    $scope.htmlData = previewHtmlData;
+                  };
 
-          $scope.openSubtopicTitleEditor = function() {
-            $scope.subtopicTitleEditorIsShown = true;
+                  $scope.save = function() {
+                    $uibModalInstance.close({
+                      newTitle: $scope.editableTitle,
+                      newHtmlData: $scope.htmlData
+                    });
+                  };
+
+                  $scope.cancel = function() {
+                    $uibModalInstance.dismiss('cancel');
+                  };
+                }
+              ]
+            });
+
+            modalInstance.result.then(function(newValues) {
+              $scope.subtopicPage = TopicEditorStateService.getSubtopicPage();
+              var newTitle = newValues.newTitle;
+              var newHtmlData = newValues.newHtmlData;
+
+              if (newTitle !== subtopic.getTitle()) {
+                TopicUpdateService.setSubtopicTitle(
+                  $scope.topic, subtopic.getId(), newTitle);
+              }
+              if (newHtmlData !== $scope.subtopicPage.getHtmlData()) {
+                TopicUpdateService.setSubtopicPageHtmlData(
+                  $scope.subtopicPage, subtopic.getId(), newHtmlData);
+                TopicEditorStateService.setSubtopicPage($scope.subtopicPage);
+              }
+            });
           };
 
-          $scope.closeSubtopicTitleEditor = function() {
-            $scope.subtopicTitleEditorIsShown = false;
+          $scope.getSkillEditorUrl = function(skillId) {
+            return UrlInterpolationService.interpolateUrl(
+              SKILL_EDITOR_URL_TEMPLATE, {
+                skillId: skillId
+              }
+            );
           };
 
-          $scope.openPreviewSubtopicPage = function(htmlData) {
-            $scope.subtopicEditorIsShown = false;
-            $scope.htmlData = htmlData;
+          /**
+           * @param {string|null} oldSubtopicId - The id of the subtopic from
+           *    which the skill is to be moved or null, if the origin is the
+           *    uncategorized section.
+           * @param {SkillSummary} skill - The summary of the skill that is to
```

Comment:
Perhaps the name of the variable should be skillSummary (and $scope.skillSummaryToMove)

Reply:
For this one I think it should be skillSummary because for these things you are actually referring to the skillSummary object here and not the conceptual skill that underlies it.

Reply:
Done.